multi-agent/lib/mas/elements/nodes/common/capabilities/agent_capable.py

```
@@ -173,7 +173,62 @@ def _ensure_executor(self, tools: List[BaseTool], config: Optional[ExecutorConfi
+                return True
+            if '"error"' in lower and ('"message"' in lower or "error:" in lower):
+                return True
+            if '"isError": true' in lower or '"is_error": true' in lower:
+                return True
+            return False
+
+        tracing = getattr(self, "_tracing", None)
+        if tracing and tracing.enabled:
+            _tool_trace_stack: dict = {}
+
+            async def tracing_pre_hook(tool, args, context):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `tracing_pre_hook`

Add return type annotation: `None`

(ANN202)